source/pdo_sqlsrv/pdo_stmt.cpp

```diff
@@ -1306,6 +1304,8 @@ sqlsrv_phptype pdo_sqlsrv_stmt::sql_type_to_php_type( SQLINTEGER sql_type, SQLUI
                        "Invalid encoding on the connection.  Must not be invalid or default." );
     }                
 
+    sqlsrv_phptype.typeinfo.encoding = local_encoding;
+
     switch( sql_type ) {
```

Comment:
it sets the encoding to stmt or conn encoding and in case of binary or invalid it is set to binary or invalid encoding respectively.